### PR TITLE
COTECH-1035 Enable no incontent player experiment via Pathfinder

### PR DESCRIPTION
**Pull request by nandy-andy**

### CODE REVIEW

File: src/core/video/no-incontent-player-experiment.ts [hunk 1]
```diff
@@ -1,27 +1,17 @@
...
-		EXPERIMENT_ID: '27342680309',
-		NO_INCONTENT_PLAYER_VARIATION_ID: '27359980218',
-	},
-	// https://app.optimizely.com/v2/projects/17632281258/experiments/27295370448/variations
-	DESKTOP: {
-		EXPERIMENT_ID: '27295370448',
-		NO_INCONTENT_PLAYER_VARIATION_ID: '27266830455',
-	},
-};
+const NO_INCONTENT_PLAYER_EXPERIMENT = defineExperiment({
```
Comment (andrzejkupczyk) [inline src/core/video/no-incontent-player-experiment.ts:5]: It's not a blocker, but perhaps adopting the camelCase naming convention would be more appropriate?

File: src/core/video/no-incontent-player-experiment.ts [hunk 2]
```diff
@@ -1,27 +1,17 @@
...
+	name: 'no-incontent-player-experiment',
+	buckets: ['A', 'B', 'C', 'D', 'E', 'F', 'G'], // 10.93%
```
Comment (andrzejkupczyk) [inline src/core/video/no-incontent-player-experiment.ts:7]: I noticed that you initially mentioned using `ABCDEF`. I just want to double-check that ☝🏻 is correct.

Comment (andrzejkupczyk) [reply src/core/video/no-incontent-player-experiment.ts]: I've updated the configuration to `ABCD` following Jase's suggestion.

File: src/core/video/no-incontent-player-experiment.ts [hunk 3]
```diff
@@ -1,27 +1,17 @@
...
-	},
-	// https://app.optimizely.com/v2/projects/17632281258/experiments/27295370448/variations
-	DESKTOP: {
-		EXPERIMENT_ID: '27295370448',
-		NO_INCONTENT_PLAYER_VARIATION_ID: '27266830455',
-	},
-};
+const NO_INCONTENT_PLAYER_EXPERIMENT = defineExperiment({
+	name: 'no-incontent-player-experiment',
+	buckets: ['A', 'B', 'C', 'D'], // 6.25%
```
Comment (latata) [inline src/core/video/no-incontent-player-experiment.ts:7]: according to https://docs.google.com/spreadsheets/d/1IxYa8v2_bl_Juquppvs4dtkjzz8TpHJZn1dXywyfBDA/edit#gid=0 we should use small letters `abcd`

File: src/platforms/shared/setup/base-context.setup.ts [hunk 4]
```diff
@@ -115,6 +115,10 @@ export class BaseContextSetup implements DiProcess {
 			'options.video.comscoreJwpTracking',
 			this.instantConfig.get('icComscoreJwpTracking'),
 		);
+		context.set(
+			'options.video.experiments.disabled.no_incontent_player',
+			this.instantConfig.get('icDisableNoIncontentPlayerExperiment'),
```
Comment (latata) [inline src/platforms/shared/setup/base-context.setup.ts:120]: minor: I'd avoid negated variables as they might be confusing in usage

File: src/core/video/no-incontent-player-experiment.ts (hunk 3 above)
Comment (andrzejkupczyk) [reply src/core/video/no-incontent-player-experiment.ts:7]: https://github.com/Wikia/pathfinder-lite?tab=readme-ov-file#default-buckets-rules
> Groups a-d are reserved for control and can never be assigned.

Comment (latata) [reply src/core/video/no-incontent-player-experiment.ts:7]: I wouldn't believe this readme ;) it is outdated probably (I assume it's from the times before we started using Optimizely) - the spreadsheet is fresh and this is our source of truth ;)

Comment (latata) [reply src/core/video/no-incontent-player-experiment.ts:7]: Also I've just realised that the spreadsheet covers only Desktop experiment. 🤔

Comment (latata) [reply src/core/video/no-incontent-player-experiment.ts:7]: Aaaah... there is another tab in the spreadsheet with mobile experiments :D

Comment (andrzejkupczyk) [reply src/core/video/no-incontent-player-experiment.ts:7]: So... Should it be `abcd` for desktop, and `CDEabc` for mobile?

Comment (latata) [reply src/core/video/no-incontent-player-experiment.ts:7]: My guess is `abcd` for desktop no player variant, and `4567` for control desktop group 
`CDE` for mobile no player variant and `abc` for mobile control group (I assume there is a mistake and `abc` was supposed to be a control group)

**Comment by nandy-andy** [comment]
> I wonder if we need a tracking event - similar to what they did here: https://github.com/Wikia/unified-platform/pull/16477/files#diff-0b8ea5c28b6a47a7561e9507b017ba493f2ef611981a00ce9a02834b8f64c17cR3 🤔

I'm not sure why it's needed if we have the information about group in the beacon. It will also make AdEngine package bigger as we'd need to add [browser-tracking-metrics](https://github.com/Wikia/browser-tracking-metrics/) to dependencies if we want to have it unified. I'd avoid it for now but we need to double-check with DA if beacon is good enough.

File: .npmrc [hunk 5]
```diff
@@ -1,2 +1,10 @@
 loglevel="warn"
 save-exact=true
+@wikia:registry=https://artifactory.wikia-inc.com/artifactory/api/npm/wikia-npm/
+@fandom:registry=https://artifactory.wikia-inc.com/artifactory/api/npm/wikia-npm/
+//artifactory.wikia-inc.com:443/artifactory/api/npm/wikia-npm/:_authToken=${ARTIFACTORY_NPM_TOKEN}
+//artifactory.wikia-inc.com:443/artifactory/api/npm/wikia-npm/:always-auth=true
+//artifactory.wikia-inc.com/artifactory/api/npm/wikia-npm/:_authToken=${ARTIFACTORY_NPM_TOKEN}
+//artifactory.wikia-inc.com/artifactory/api/npm/wikia-npm/:always-auth=true
```
Comment (latata) [inline .npmrc:8]: This is not needed anymore, right? 🤔

File: src/core/video/no-incontent-player-experiment.ts [hunk 6]
```diff
@@ -1,27 +1,65 @@
+import defineExperiment from '@fandom/pathfinder-lite/experiments/defineExperiment';
```
Comment (latata) [inline src/core/video/no-incontent-player-experiment.ts:1]: `@wikia/pathfinder-lite`

File: .npmrc (hunk 5 above)
Comment (andrzejkupczyk) [reply .npmrc:8]: Indeed

**Review by dzawada — DISMISSED** [review]

**Review by dzawada — APPROVED** [review]

**Review by latata — APPROVED** [review]

**Pull request merged**